go/libraries/doltcore/sqle/binlogreplication/binlog_replication_test.go

```diff
@@ -473,6 +475,45 @@ func waitForReplicaToCatchUp(t *testing.T) {
 	t.Fatal("primary and replica did not synchronize within " + timeLimit.String())
 }
 
+// waitForReplicaToHaveLatestGtid waits (up to 60s) for the replica to contain the
+// most recent GTID executed from the primary. Both the primary and replica are queried
+// for the value of @@gtid_executed to determine if the replica contains the most recent
+// transaction from the primary.
+func waitForReplicaToHaveLatestGtid(t *testing.T) {
+	timeLimit := 60 * time.Second
```

Comment:
This seems pretty excessive. What's the actual usual lag?

Reply:
Yeah, this can be tuned down. If this function doesn't find the latest GTID within 10s, then it's not worth waiting any longer.